__tests__/integration/atomic-swap.test.js

```diff
@@ -0,0 +1,900 @@
+    let response = await TestUtils.request
+      .get('/wallet/transaction')
+      .query({ id: tx.hash })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[change]: check fundsTx', { body: response.body });
+    expect(response.body.tx_id).toBe(tx.hash);
+
+    for (const [index, output] of Object.entries(response.body.outputs)) {
+      if (output.decoded.address === destAddr) {
+        fundsIndex = +index; // convert to number
+        fundsOutput = output;
+        break;
+      }
+    }
+    loggers.test.insertLineToLog('atomic-swap[change]: funds utxo', { index: fundsIndex, output: fundsOutput });
+    expect(typeof fundsIndex).toBe('number');
+    expect(fundsOutput).toMatchObject({
+      token: '00',
+      value: expect.any(Number),
+      decoded: expect.objectContaining({
+        address: destAddr,
+      }),
+    });
+
+    // Send 1 HTR and creating 2 HTR of change
+
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        send: {
+          tokens: [{ value: 1 }],
+          utxos: [{ txId: tx.hash, index: fundsIndex }],
+        },
+        change_address: changeAddr,
+        lock: false,
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[change]: proposal send change', { body: response.body });
+
+    let partialTx = PartialTx.deserialize(response.body.data, network);
+    // Prepare outputs to be checked
+    for (const output of partialTx.outputs) {
+      output.parseScript(network);
+    }
+
+    expect(partialTx).toMatchObject({
+      inputs: [expect.objectContaining({ // only 1 input
+        hash: tx.hash,
+        index: fundsIndex,
+      })],
+      outputs: [expect.objectContaining({ // only 1 output
+        isChange: true,
+        token: '00',
+        value: 2,
+        authorities: 0,
+        decodedScript: expect.objectContaining({
+          address: expect.objectContaining({ base58: changeAddr }),
+        }),
+      })],
+    });
+
+    // Receive HTR specifying the address
+
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        receive: {
+          tokens: [{ value: 4, address: recvAddr }],
+        },
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[change]: proposal receive change', { body: response.body });
+
+    partialTx = PartialTx.deserialize(response.body.data, network);
+    // Prepare outputs to be checked
+    for (const output of partialTx.outputs) {
+      output.parseScript(network);
+    }
+    expect(partialTx).toMatchObject({
+      outputs: [expect.objectContaining({
+        isChange: false,
+        token: '00',
+        value: 4,
+        authorities: 0,
+        decodedScript: expect.objectContaining({
+          address: expect.objectContaining({ base58: recvAddr }),
+        }),
+      })],
+    });
+  });
```

Comment:
I would only add one more test here:

1. Create a proposal in a way that a change output must be created
2. Don't select the change address
3. Validate that the automatically selected change address belongs to the wallet correctly

Reply:
Done